presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -721,19 +721,24 @@ public static long levenshteinDistance(@SqlType("varchar(x)") Slice left, @SqlTy
     @SqlType(StandardTypes.BIGINT)
     public static long hammingDistance(@SqlType("varchar(x)") Slice left, @SqlType("varchar(y)") Slice right)
     {
-        int[] leftCodePoints = castToCodePoints(left);
-        int[] rightCodePoints = castToCodePoints(right);
-
         checkCondition(
-                leftCodePoints.length == rightCodePoints.length,
+                safeCountCodePoints(left) == safeCountCodePoints(right),
                 INVALID_FUNCTION_ARGUMENT,
-                "The input strings for Hamming distance must have the same length");
+                "The input strings to hamming_distance function must have the same length");
 
         int distance = 0;
-        for (int i = 0; i < leftCodePoints.length; i++) {
-            if (leftCodePoints[i] != rightCodePoints[i]) {
+        int leftPosition = 0;
+        int rightPosition = 0;
+        while (leftPosition < left.length()) {
```

Comment:
Thanks for the review! My understanding is that we could ignore the invalid characters and calculate the result as is (garbage) to avoid failure. One question, does -5 and -6 for codePoint mean it could safely move the pointer for 5 and 6 characters?